_test/common_functions/is_cut_equal.m

```diff
@@ -12,48 +14,85 @@
-end
-for i=1:numel(f2)
-    w2(i)=cut_sqw(f2{i},varargin{:});
-end
+    [f1, f2, proj, pN, cutArgs, tol] = parse_args(varargin{:});
 
-w1tot=combine_cuts(w1);
-w2tot=combine_cuts(w2);
-
-% To check equality, see if npix, s, e arrays are the same
-if equal_to_tol(w1tot.data.npix,w2tot.data.npix) &&...
-        equal_to_tol(w1tot.data.s,w2tot.data.s) &&...
-        equal_to_tol(w1tot.data.e,w2tot.data.e) &&...
-        equal_to_tol(w1tot.data.img_range,w2tot.data.img_range)
-    if isempty(w1tot.data.pix)
-        ok=true;
-        mess='';
-    else
-        if equal_to_tol(w1tot.data.pix.pix_range,w2tot.data.pix.pix_range) && ...
-                equal_to_tol(w1tot.data.pix.num_pixels,w2tot.data.pix.num_pixels)
+    w1=repmat(sqw,1,numel(f1));
+    w2=repmat(sqw,1,numel(f2));
+    for i=1:numel(f1)
+        w1(i)=cut_sqw(f1{i},proj,pN{:},cutArgs{:});
+    end
+    for i=1:numel(f2)
+        w2(i)=cut_sqw(f2{i},proj,pN{:},cutArgs{:});
+    end
+
+    w1tot=combine_cuts(w1);
+    w2tot=combine_cuts(w2);
+
+    % To check equality, see if npix, s, e arrays are the same
+    if equal_to_tol(w1tot.data.npix,w2tot.data.npix,tol) &&...
+            equal_to_tol(w1tot.data.s,w2tot.data.s,tol) &&...
+            equal_to_tol(w1tot.data.e,w2tot.data.e,tol) &&...
+            equal_to_tol(w1tot.data.img_range,w2tot.data.img_range,tol)
+        if isempty(w1tot.data.pix)
             ok=true;
             mess='';
         else
-            ok = false;
-            mess=sprintf(['Pixels parameters of two cuts are different:\n',...
-                '   npix1=%d and npix2 = %d\n',...
-                'pix_range1 = [%f  %f %f %f; pix_range2=[%f  %f %f  %f;\n',...
-                '              %f  %f %f %f]             %f  %f %f  %f]'],...
-                w1tot.data.pix.num_pixels,w2tot.data.pix.pix_range,...
-                w1tot.data.pix.pix_range(1,:),w2tot.data.pix.pix_range(1,:),...
-                w1tot.data.pix.pix_range(2,:),w2tot.data.pix.pix_range(2,:));
+            if equal_to_tol(w1tot.data.pix.pix_range,w2tot.data.pix.pix_range,tol) && ...
+                    equal_to_tol(w1tot.data.pix.num_pixels,w2tot.data.pix.num_pixels,tol)
+                ok=true;
+                mess='';
+            else
+                ok = false;
+                mess=sprintf(['Pixels parameters of two cuts are different:\n',...
+                              '   npix1=%d and npix2 = %d\n',...
+                              'pix_range1 = [%f  %f %f %f; pix_range2=[%f  %f %f  %f;\n',...
+                              '              %f  %f %f %f]             %f  %f %f  %f]'],...
+                             w1tot.data.pix.num_pixels,w2tot.data.pix.pix_range,...
+                             w1tot.data.pix.pix_range(1,:),w2tot.data.pix.pix_range(1,:),...
+                             w1tot.data.pix.pix_range(2,:),w2tot.data.pix.pix_range(2,:));
+            end
         end
+    else
+        ok=false;
+        mess='One or more of npix, s, e are not the same';
```

Comment:
the detail of the two not-equal messages is wildly different: array elements vs one of these four variables.
Can more detail be added here to name which is/are not equal?